oceanspy/utils.py

```diff
@@ -520,7 +520,7 @@ def densjmd95(s, t, p):
         if isinstance(var, _xr.DataArray):
             var = var.astype("float")
         else:
-            var = _np.asfarray(var)
+            var = _np.asarray(var, _np.float64)
```

Comment:
I am not so sure about this. There are two parts to this snippet:

a) Unchanged: If the file is an xr.DataArray (line 520) it probably has in its attributes what type it is. And so simply casting it as a float may not be the best idea. For example it may be an integer. We should probably respect the original type.


b) Line 523: In this case it is not an xarray datarray, which means there may not have attributes associated with it. decided to cast it as float64, but this seems arbitrary. We could make make it float 32. And so this ambiguity made me paused.

Reply:
Maybe raise a warning at this point?

Reply:
I decided to simply cast it as float32. I looked at this a bit closer, and this function computes density from S, T and P. All these are floats (actually float32). I am marking this as resolved and will merge this PR